docs/02.console_producers_and_consumers.md

```diff
@@ -0,0 +1,94 @@
+### 1. Create a topic before producing/consuming messages from Kafka.
+```docker exec kafka-workshop-1 kafka-topics --create --bootstrap-server localhost:29092 --partitions 3 --replication-factor 3 --topic my-first-topic```
+
+### 2.  Verify topic creation or describe the topic
+```docker exec kafka-workshop-1 kafka-topics --bootstrap-server localhost:29092 --describe --topic my-first-topic```
+
+### 3.  Produce messages to newly created Kafka topic
+
+```shell script
+docker exec --interactive --tty kafka-workshop-1 kafka-console-producer --bootstrap-server localhost:29092 --topic my-first-topic
+```
+### 4.  Read messages from Kafka topic
+
+```shell script
+docker exec --interactive --tty kafka-workshop-1 kafka-console-consumer --bootstrap-server localhost:29092 --topic my-first-topic --from-beginning
+```
+
+### 5.  Other commands to try while reading messages from Kafka
+
+- print partition and offset details
+```shell script
+docker exec --interactive --tty kafka-workshop-1 kafka-console-consumer --bootstrap-server localhost:29092 --topic my-first-topic --from-beginning --property print.partition=true --property print.offset=true
+```
+
+- read from specific partition
+```shell script
+docker exec --interactive --tty kafka-workshop-1 kafka-console-consumer --bootstrap-server localhost:29092 --topic my-first-topic --partition 0 --from-beginning
+```
+
+- read from specific partition & offset
+```shell script
+docker exec --interactive --tty kafka-workshop-1 kafka-console-consumer --bootstrap-server localhost:29092 --topic my-first-topic --partition 2 --offset 1
+```
+
+#### Updating an existing topics:
+
+The configuration of existing topics can be updates. Below is one example
+
+- update the min.insync.replicas config of the topic
+```shell script
+docker exec kafka-workshop-1 kafka-configs --bootstrap-server localhost:29092 --alter --entity-type topics --entity-name my-first-topic --add-config min.insync.replicas=3
+```
+
+**********************
+### 1.  Create a topic first beore producing/consuming messages from Kafka.
```

Comment:
can we run commands in this section without docker exec ?

Reply:
can we check this together? We want to run the commands on the docker container created in previous step. Why it was not able to connect for you, we can check.